cpp/src/arrow/flight/sql/odbc/odbcabstraction/odbc_impl/odbc_statement.cc

```diff
@@ -737,6 +740,31 @@ bool ODBCStatement::GetData(SQLSMALLINT recordNumber, SQLSMALLINT cType,
                                  bufferLength, indicatorPtr);
 }
 
+SQLRETURN ODBCStatement::getMoreResults() {
+  // Multiple result sets are not supported.
+  if (m_currenResult) {
+    return SQL_NO_DATA;
+  } else {
+    throw DriverException("Function sequence error", "HY010");
+  }
+}
+
+void ODBCStatement::getColumnCount(SQLSMALLINT* columnCountPtr) {
+  if (!columnCountPtr) {
+    throw DriverException("Invalid null pointer for number of column results.", "HY000");
+  }
+  size_t columnCount = m_currentArd->GetRecords().size();
+  *columnCountPtr = columnCount;
+}
+
+void ODBCStatement::getRowCount(SQLLEN* rowCountPtr) {
+  if (!rowCountPtr) {
```

Comment:
I don't see in the spec this error mentioned, which means we can actually just return success and not do anything. Same with num result cols.